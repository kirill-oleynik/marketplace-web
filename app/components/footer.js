import React from 'react';
import { Container } from 'reactstrap';

class MainFooter extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      modal: false
    };
    this.toggle = this.toggle.bind(this);
  }

  toggle() {
    this.setState({
      modal: !this.state.modal
    });
  }

  render() {
    return (
      <footer>
        <Container>
          <div className="main-footer main-footer-js">
            <div className="main-footer__wrap">
              <a href="/" className="page-main-logo in-blue-500">Appreview</a>
              <p className="font-14 in-black-035 ml-10 mb-0">Â©2017 All Rights Reserved</p>
            </div>
            <div className="main-footer__wrap">
              <a href="/" className="main-footer__link">Terms of Use</a>
              <a href="/" className="main-footer__link">Privacy Policy</a>
            </div>
          </div>
        </Container>
      </footer>
    );
  }
}

export default MainFooter;
