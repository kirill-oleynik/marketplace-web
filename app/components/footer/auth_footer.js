import React from 'react';

import PropTypes from 'prop-types';
import { translate } from 'react-i18next';

import { getCurrentYear } from '../../helpers/dates_helpers';

export const AuthFooter = ({ t, onTermsOfUseClick, onPrivacyPolicyClick }) => (
  <footer>
    <p className="font-12 in-black-035 mb-0">
      {t('footer.applicationName')}

      &nbsp;

      {t('footer.copyright', { year: getCurrentYear() })}
    </p>

    <p className="font-12 in-black-035 mb-0">
      <span className="rights-link rights-link__privacy" onClick={onPrivacyPolicyClick}>
        {t('footer.privacyPolicy')}
      </span>

      &nbsp;{t('footer.and')}&nbsp;

      <span className="rights-link rights-link__terms" onClick={onTermsOfUseClick}>
        {t('footer.termsOfUse')}
      </span>
    </p>
  </footer>
);

AuthFooter.propTypes = {
  t: PropTypes.func.isRequired,
  onTermsOfUseClick: PropTypes.func.isRequired,
  onPrivacyPolicyClick: PropTypes.func.isRequired
};

export default translate(['common'])(AuthFooter);
