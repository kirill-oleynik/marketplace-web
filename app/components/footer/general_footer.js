import React from 'react';
import Link from 'next/link';
import PropTypes from 'prop-types';
import { Container } from 'reactstrap';
import { translate } from 'react-i18next';

import { home } from '../../routes';
import { getCurrentYear } from '../../helpers/dates_helpers';

export const GeneralFooter = ({ t, onTermsOfUseClick, onPrivacyPolicyClick }) => (
  <footer>
    <Container>
      <div className="main-footer main-footer-js">
        <div className="main-footer__wrap">
          <Link href={home}>
            <a className="page-main-logo in-blue-500">
              {t('footer.applicationName')}
            </a>
          </Link>

          <p className="font-14 in-black-035 ml-10 mb-0">
            {t('footer.copyright', { year: getCurrentYear() })}
          </p>
        </div>

        <div className="main-footer__wrap">
          <span
            onClick={onTermsOfUseClick}
            className="main-footer__link main-footer__terms-link"
          >
            {t('footer.termsOfUse')}
          </span>

          <span
            onClick={onPrivacyPolicyClick}
            className="main-footer__link main-footer__privacy-link"
          >
            {t('footer.privacyPolicy')}
          </span>
        </div>
      </div>
    </Container>
  </footer>
);

GeneralFooter.propTypes = {
  t: PropTypes.func.isRequired,
  onTermsOfUseClick: PropTypes.func.isRequired,
  onPrivacyPolicyClick: PropTypes.func.isRequired
};

export default translate(['common'])(GeneralFooter);
