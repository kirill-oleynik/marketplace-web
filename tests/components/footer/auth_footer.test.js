import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { AuthFooter } from 'app/components/footer/auth_footer';

const onTermsOfUseClickSpy = jest.fn();
const onPrivacyPolicyClickSpy = jest.fn();

const renderFooter = () => (
  <AuthFooter
    t={(translation) => translation}
    onTermsOfUseClick={onTermsOfUseClickSpy}
    onPrivacyPolicyClick={onPrivacyPolicyClickSpy}
  />
);

describe('#render', () => {
  test('it renders correctly', () => {
    const component = shallow(
      renderFooter()
    );

    expect(
      toJSON(component)
    ).toMatchSnapshot();
  });
});

describe('on Terms Of Use click', () => {
  it('calls onTermsOfUseClick', () => {
    const component = shallow(
      renderFooter()
    );

    component.find('.rights-link__terms').simulate('click');

    expect(onTermsOfUseClickSpy).toHaveBeenCalled();

    onTermsOfUseClickSpy.mockReset();
    onTermsOfUseClickSpy.mockRestore();
  });
});

describe('on Privacy Policy click', () => {
  it('calls onPrivacyPolicyClick', () => {
    const component = shallow(
      renderFooter()
    );

    component.find('.rights-link__privacy').simulate('click');

    expect(onPrivacyPolicyClickSpy).toHaveBeenCalled();

    onPrivacyPolicyClickSpy.mockReset();
    onPrivacyPolicyClickSpy.mockRestore();
  });
});
