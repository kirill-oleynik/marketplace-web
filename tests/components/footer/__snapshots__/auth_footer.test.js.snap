// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render it renders correctly 1`] = `
<footer>
  <p
    className="font-12 in-black-035 mb-0"
  >
    footer.applicationName
     
    footer.copyright
  </p>
  <p
    className="font-12 in-black-035 mb-0"
  >
    <span
      className="rights-link rights-link__privacy"
      onClick={[Function]}
    >
      footer.privacyPolicy
    </span>
     
    footer.and
     
    <span
      className="rights-link rights-link__terms"
      onClick={[Function]}
    >
      footer.termsOfUse
    </span>
  </p>
</footer>
`;
